services/src/main/java/org/keycloak/authentication/AuthenticationSelectionResolver.java

```diff
@@ -91,11 +92,24 @@
                         .map(credentialType -> new AuthenticationSelectionOption(processor.getSession(), typeAuthExecMap.get(credentialType)))
                         .collect(Collectors.toList());
             }
+            else {
+                // No user associated with session. Check if this flow contains executions linked to authenticators that don't require a user
+                typeAuthExecMap.forEach((key, value) -> {
+                    AuthenticatorFactory credbasedAuthenticatorFactory = (AuthenticatorFactory) processor.getSession().getKeycloakSessionFactory().getProviderFactory(Authenticator.class, value.getAuthenticator());
```

Comment:
I was wondering if it's a proper solution to obtain `sessionFactory` in each iteration as the session (AFAIK) should be the same for the whole method. However, as one of the main purposes of the `AuthenticationProcessor` class is to manage the state of the authentication process and there are no other calls to dynamically changing objects, I don't consider this as a blocker for this PR. Just wanted to take it into consideration.